test/e2e/framework/kcp.go

```diff
@@ -236,7 +236,11 @@ func (c *kcpServer) Run(opts ...RunOption) error {
 
 	// run kcp start in-process for easier debugging
 	if runOpts.runInProcess {
-		serverOptions := options.NewOptions(".kcp")
+		rootDir := ".kcp"
+		if c.dataDir != "" {
+			rootDir = c.dataDir
+		}
+		serverOptions := options.NewOptions(rootDir)
```

Comment:
self-explanatory comment: this effectively moves the root directory to a temporary directory if running as in-process, maybe worth a small comment (not a blocker for merging).